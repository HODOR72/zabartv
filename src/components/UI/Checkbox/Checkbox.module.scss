.checkbox {
	position: relative;
	display: inline-flex;
	align-items: center;

	@media (max-width: 576px) {
		align-items: flex-start;
	}
}

.switch {
	position: relative;
	display: block;
	width: 24px;
	height: 24px;
	flex-shrink: 0;
	margin-right: 10px;

	&:before {
		content: '';
		display: block;
		position: absolute;
		top: 0;
		left: 0;
		right: 0;
		bottom: 0;
		border: 1px solid #38445b;
		border-radius: 7px;
		transition: opacity 0.3s, visibility 0.3s, border-color 0.3s;
	}

	&:after {
		content: '';
		display: block;
		position: absolute;
		top: 0;
		left: 0;
		right: 0;
		bottom: 0;
		opacity: 0;
		visibility: hidden;
		transition: opacity 0.3s, visibility 0.3s;
		background-image: url("data:image/svg+xml,%3Csvg width='24' height='24' viewBox='0 0 24 24' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M17.0365 0H6.97548C2.6053 0 0 2.604 0 6.972V17.016C0 21.396 2.6053 24 6.97548 24H17.0245C21.3947 24 24 21.396 24 17.028V6.972C24.012 2.604 21.4067 0 17.0365 0ZM17.7448 9.24L10.9374 16.044C10.7694 16.212 10.5413 16.308 10.3011 16.308C10.061 16.308 9.8329 16.212 9.66482 16.044L6.26712 12.648C5.91895 12.3 5.91895 11.724 6.26712 11.376C6.6153 11.028 7.19158 11.028 7.53976 11.376L10.3011 14.136L16.4722 7.968C16.8204 7.62 17.3967 7.62 17.7448 7.968C18.093 8.316 18.093 8.88 17.7448 9.24Z' fill='%23FCA11A'/%3E%3C/svg%3E%0A");
	}
}

[data-theme='light'] {
	.switch:before {
		border-color: #e2e0dd;
	}
}

.input {
	-webkit-appearance: none;
	appearance: none;
	outline: none;
	border: none;
	position: absolute;
	z-index: 1;
	width: 100%;
	height: 100%;
	background-color: transparent;
	cursor: pointer;

	&:checked ~ .switch {
		&:before {
			opacity: 0;
			visibility: visible;
		}

		&:after {
			opacity: 1;
			visibility: visible;
		}
	}
}

.label {
	font-size: 16px;
	line-height: 28.8px;
	font-weight: 500;
	color: var(--color-white);
	transition: color 0.3s;

	@media (max-width: 576px) {
		font-size: 14px;
		line-height: 24px;
	}
}
