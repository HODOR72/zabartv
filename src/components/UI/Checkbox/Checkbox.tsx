import classNames from 'classnames';
import { FC, InputHTMLAttributes } from 'react';
import styles from './Checkbox.module.scss';

interface CheckboxProps extends InputHTMLAttributes<HTMLInputElement> {
	label?: string;
}

export const Checkbox: FC<CheckboxProps> = ({ label, className, ...props }) => {
	return (
		<label className={classNames(styles.checkbox, className)}>
			<input data-testid="checkbox-input" className={styles.input} type="checkbox" {...props} />
			<span className={styles.switch} />
			{label && <span className={styles.label}>{label}</span>}
		</label>
	);
};
