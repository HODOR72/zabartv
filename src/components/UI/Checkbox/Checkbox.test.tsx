import { render, screen, fireEvent } from '@testing-library/react';
import { Checkbox } from './Checkbox';

describe('Checkbox.test.tsx', () => {
	test('if checked', () => {
		render(<Checkbox data-testid="checkbox-input" />);
		const element = screen.getByTestId('checkbox-input');

		expect(element).not.toBeChecked();

		fireEvent.click(element);
		expect(element).toBeChecked();

		expect(element).toMatchSnapshot();
	});
});
