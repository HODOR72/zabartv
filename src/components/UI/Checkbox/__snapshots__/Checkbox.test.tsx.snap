// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox.test.tsx if checked 1`] = `
<input
  class="input"
  data-testid="checkbox-input"
  type="checkbox"
/>
`;
